cmd/stats/stats_test.go

```diff
@@ -0,0 +1,395 @@
+	nowBeforeMidnight := nowAtMidnight.AddDate(0, 0, 1).Add(-1 * time.Nanosecond)
+
+	type testCase struct {
+		name    string
+		in      []string
+		want    string
+		wantErr bool
+	}
+
+	var mockCmd = func(tc testCase) *cobra.Command {
+		cmd := &cobra.Command{
+			Run: func(cmd *cobra.Command, args []string) {
+				got, err := argsToQuery(cmd, now)
+				if err == nil && tc.wantErr {
+					t.Errorf("%s wanted error, got nil", tc.name)
+				}
+
+				if got != tc.want {
+					t.Errorf("%s\n"+
+						"  got:  %s\n"+
+						"  want: %s\n",
+						tc.name, got, tc.want)
+				}
+			},
+		}
+		setStatsCommandFlags(cmd)
+		cmd.SetArgs(tc.in)
+		return cmd
+	}
+
+	testCases := []testCase{
+		{
+			name: "default case (get stats from today only)",
+			in:   []string{},
+			want: fmt.Sprintf(
+				"select * from reps where start >= %d and end <= %d order by start desc;",
+				nowAtMidnight.UnixMilli(),
+				nowBeforeMidnight.UnixMilli(),
+			),
+			wantErr: false,
+		},
+		{
+			name: "since is an alias for start",
+			in:   []string{"--since=2D"},
+			want: fmt.Sprintf(
+				"select * from reps where start >= %d and end <= %d order by start desc;",
+				nowAtMidnight.AddDate(0, 0, -2).UnixMilli(),
+				nowBeforeMidnight.UnixMilli(),
+			),
+			wantErr: false,
+		},
+		{
+			name: "both since and start are given: warn, and prefer start",
+			in:   []string{"--since=2D", "--start=1D"},
+			want: fmt.Sprintf(
+				"select * from reps where start >= %d and end <= %d order by start desc;",
+				nowAtMidnight.AddDate(0, 0, -1).UnixMilli(),
+				nowBeforeMidnight.UnixMilli(),
+			),
+			wantErr: false,
+		},
+		{
+			name: "start provided",
+			in:   []string{"--start=1D"},
+			want: fmt.Sprintf(
+				"select * from reps where start >= %d and end <= %d order by start desc;",
+				nowAtMidnight.AddDate(0, 0, -1).UnixMilli(),
+				nowBeforeMidnight.UnixMilli(),
+			),
+			wantErr: false,
+		},
+		{
+			name:    "end provided, but no start",
+			in:      []string{"--end=1D"},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name: "start and end provided",
+			in:   []string{"--start=2024-10-01", "--end=2024-11-01"},
+			want: fmt.Sprintf(
+				"select * from reps where start >= %d and end <= %d order by start desc;",
+				time.Date(2024, time.October, 1, 0, 0, 0, 0, now.Location()).UnixMilli(),
+				time.Date(2024, time.November, 1, 0, 0, 0, 0, now.Location()).AddDate(0, 0, 1).Add(-1*time.Nanosecond).UnixMilli()),
+			wantErr: false,
+		},
+		{
+			name:    "start and end provided, but end is before start",
+			in:      []string{"--start=1D", "--end=3D"},
+			want:    "",
+			wantErr: true,
+		},
+		{
+			name: "language provided",
+			in:   []string{"--lang=py"},
+			want: fmt.Sprintf(
+				"select * from reps where lang='py' and start >= %d and end <= %d order by start desc;",
+				nowAtMidnight.UnixMilli(),
+				nowBeforeMidnight.UnixMilli(),
+			),
+			wantErr: false,
+		},
+		{
+			name: "name provided",
```

Comment:
I just realized that getting the stats of a corresponding name is gonna be more complicated than I thought.

If I want to fetch reps by name in the database, what I'm really doing is fetching reps by _file_ (or, by the MD5 hash of their contents) in the database.

Suppose I do a couple exercises. One is in the exercises directory called `hello`, and another is a file from the current directory called `hello`. Suppose their contents are different too. When I'm looking for stats of a file matching the name `hello`, I only want to see entries for _one of those files_. Comparing stats for files of different contents is not a useful comparison, especially if the files are wildly different.

So how to tackle this? There are some options.
- Add a `hash` flag to only match files by their hash
- Add a `--file` flag, which accepts a path to a file, and then queries the table by the hash of the contents of the file?
  - would this make the `--name` flag unnecessary then?
- Pass a file path as an argument to the stats command, which does the same as the proposed `--file` flag?
  - consistent behavior to the `sweet` command, which I like

I may have to put this in it's own issue and address it later, depending on how people use the original sweet command

Reply:
See #74